central/detection/service/service_impl.go

```diff
@@ -340,11 +351,41 @@ func (s *serviceImpl) DetectDeployTimeFromYAML(ctx context.Context, req *apiV1.D
 		eCtx.ClusterID = clusterID
 	}
 
+	var deployments []*storage.Deployment
+	var errs []error
+
 	var runs []*apiV1.DeployDetectionResponse_Run
 	for _, r := range resources {
-		run, err := s.runDeployTimeDetect(ctx, eCtx, r, req.GetPolicyCategories())
+		d, err := s.convertK8sResource(r)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		deployments = append(deployments, d)
+	}
+	if len(deployments) == 0 && len(errs) > 0 {
+		msg := fmt.Sprintf("every deployment YAML failed to parse - aborting. Errors:\n%v", errs)
+		return nil, errox.InvalidArgs.New(msg)
+	}
+	if len(deployments) > 0 && len(errs) > 0 {
+		log.Warnf("one or multiple deployment YAMLs failed to parse:\n%v", errs)
+	}
+
+	// Enhance the deployments, then range over them
+	conn := s.connManager.GetConnection(eCtx.ClusterID)
+	if conn == nil {
+		return nil, errox.InvalidArgs.New("Connection to cluster is not ready. Try again in a few seconds.")
+	}
+	enhancedDeployments, err := s.enhancementWatcher.SendAndWaitForEnhancedDeployments(ctx, conn, deployments, 30*time.Second)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed waiting for augmented deployment response")
+	}
+
+	for _, d := range enhancedDeployments {
+		run, err := s.runDeployTimeDetect(ctx, eCtx, d, req.GetPolicyCategories())
+
 		if err != nil {
-			return nil, errox.InvalidArgs.New("unable to convert object").CausedBy(err)
+			return nil, errox.InvalidArgs.New("unable to add additional information to deployment").CausedBy(err)
```

Comment:
Please unify the errors, so that they all start with capital or non-capital letter.

Reply:
Done!